sljit_src/sljitNativeX86_common.c

```diff
@@ -423,32 +423,24 @@ static void execute_cpu_id(sljit_u32 info[4])
 	/* AT&T syntax. */
 	__asm__ (
 #if (defined SLJIT_CONFIG_X86_32 && SLJIT_CONFIG_X86_32)
-		"movl %0, %%esi\n"
-		"movl (%%esi), %%eax\n"
-		"movl 8(%%esi), %%ecx\n"
 		"pushl %%ebx\n"
+#endif /* SLJIT_CONFIG_X86_32 */
+		"movl (%0), %%eax\n"
```

Comment:
How do you ensure, that `%0` is not `eax`-`edx` ?

Reply:
eax, ecx and edx are not used because of the clobber, it could be ebx which then looks really funny, because it is first pushed to the stack by the epilogue and then again by the inlined assembler.

Reply:
I don't know what is the best. But it looks like we cannot trust in the compiler too much. Somehow we can create fully asm functions, maybe that is the best.